### ISSUE TITLE
MailerLite API is pretty broken atm

### ISSUE THREAD

**Issue by chiperific** [jpalumickas/mailerlite-ruby#38]
First, thank you so much for this gem. It's way better than my HTTParty-based client.

I've exchanged a few emails with MailerLite about their "Classic" API and several endpoints are broken. I know `webhooks` and `campaigns_by_status` specifically return 404. And `subscriber_activities` is returning a 500 error.

Even though the "Classic" API has a V1 and V2, they say they are launching a V2 API and pointed me to new docs:

> Here’s the documentation for the New version, some endpoints from the Classic are not available yet: https://mailerlite.github.io/reference/#mailerlite-api

They change the base URL to `https://connect.mailerlite.com/api` and have only rolled out some of the endpoints, with no ETA for completion, even for specific milestones like `webhooks`.

This is mostly just a heads-up that their current API is pretty broken and they don't seem to be maintaining it while they work on V2.

If you're still willing to maintain this repo, I'd love to help, maybe add in support for the new API with fallbacks to the old.

**Comment by jpalumickas** [comment]
hey @chiperific 

Hmm, I see, so as I understand now we need to change some endpoints to the new one, but while they not completed v2 API fully, we need support v1 (with v2 version) api, right?

**Comment by chiperific** [comment]
Yes, my thought was to add some checks on initialization. Try the new endpoints, if they return 404, then fall back to the old.

I think this would mean having separate host variables for the main endpoints: subscribers, groups, webhooks, campaigns, etc.

Maybe that's overkill though. We could just change the methods we know are now available in V2 and then just try to watch their blog for more releases.

**Comment by chiperific** [comment]
Yes, my though was to add some checks on initialization. Try the new
endpoints, if they return 404, then fall back to the old.

I think this would mean having separate host variables for the main
endpoints: subscribers, groups, webhooks, campaigns, etc.

Maybe that's overkill though. We could just change the methods we know are
now available in V2 and then just try to watch their blog for more releases.

- Chip

**Comment by samsonjs** [comment]
It looks like the new API can [filter campaigns by status](https://developers.mailerlite.com/docs/campaigns.html#campaign-list) and show [subscriber activity](https://developers.mailerlite.com/docs/campaigns.html#get-subscribers-activity-of-a-sent-campaign) so they've been making progress. I don't see anything about webhooks yet though.

**Comment by jpalumickas** [comment]
@samsonjs nice! let's wait little bit more. Is v1 still broken?

**Comment by samsonjs** [comment]
@jpalumickas We sometimes get 5xx level errors but overall v1 works fine for us, and there's no indication that the errors we do see would be fixed by using the new API.
